Mac OS Intel build failure:
```
FAILED: modules/imgproc/CMakeFiles/opencv_imgproc.dir/accum.avx.cpp.o 
/usr/local/bin/ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DCVAPI_EXPORTS -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DCV_CPU_COMPILE_AVX=1 -DCV_CPU_COMPILE_POPCNT=1 -DCV_CPU_COMPILE_SSE4_2=1 -DCV_CPU_DISPATCH_MODE=AVX -I/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/hal/ipp/include -I3rdparty/ippicv/ippicv_mac/icv/include -I3rdparty/ippicv/ippicv_mac/iw/include -I/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/3rdparty/dlpack/include -I/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/modules/imgproc/include -Imodules/imgproc -I/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/modules/core/include -isystem . -isystem /usr/local/include/eigen3 -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -fdiagnostics-show-option -Qunused-arguments -Wno-semicolon-before-method-body -ffunction-sections -fdata-sections  -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -mmacosx-version-min=12.0 -fPIC -isystem/usr/local/include -std=c++11  -mpopcnt -msse4.2 -mavx -MD -MT modules/imgproc/CMakeFiles/opencv_imgproc.dir/accum.avx.cpp.o -MF modules/imgproc/CMakeFiles/opencv_imgproc.dir/accum.avx.cpp.o.d -o modules/imgproc/CMakeFiles/opencv_imgproc.dir/accum.avx.cpp.o -c modules/imgproc/accum.avx.cpp
In file included from modules/imgproc/accum.avx.cpp:3:
/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/modules/imgproc/src/accum.simd.hpp:3054:58: error: always_inline function '_mm256_cvtepi8_epi32' requires target feature 'avx2', but would be inlined into function 'accW_simd_' that is compiled without support for 'avx2'
            __m256 mf = _mm256_cmp_ps(_mm256_cvtepi32_ps(_mm256_cvtepi8_epi32(mb)), v_zero, _CMP_NEQ_OQ);
                                                         ^
/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/modules/imgproc/src/accum.simd.hpp:3386:65: error: always_inline function '_mm256_cvtepi8_epi32' requires target feature 'avx2', but would be inlined into function 'accW_simd_' that is compiled without support for 'avx2'
            __m256d mf = _mm256_castsi256_pd(_mm256_cmpgt_epi32(_mm256_cvtepi8_epi32(mb), _mm256_setzero_si256()));
                                                                ^
/Users/opencv-cn/GHA-OCV-5/_work/opencv/opencv/opencv/modules/imgproc/src/accum.simd.hpp:3386:46: error: always_inline function '_mm256_cmpgt_epi32' requires target feature 'avx2', but would be inlined into function 'accW_simd_' that is compiled without support for 'avx2'
            __m256d mf = _mm256_castsi256_pd(_mm256_cmpgt_epi32(_mm256_cvtepi8_epi32(mb), _mm256_setzero_si256()));
                                             ^
3 errors generated.
```

--- a/modules/imgproc/src/accum.simd.hpp
+++ b/modules/imgproc/src/accum.simd.hpp
@@ -3034,7 +3034,9 @@ void accW_simd_(const float* src, float* dst, const uchar* mask, int len, int cn
     {
     const __m256 v_alpha = _mm256_set1_ps((float)alpha);
     const __m256 v_beta = _mm256_set1_ps((float)(1.0f - alpha));
+#if CV_AVX2
     const __m256 v_zero = _mm256_setzero_ps();
+#endif
     const int ymmWidth = 16;
 
     if (!mask)
@@ -3051,7 +3053,14 @@ void accW_simd_(const float* src, float* dst, const uchar* mask, int len, int cn
         for ( ; x <= len - 8; x += 8)
         {
             __m128i mb = _mm_loadl_epi64((const __m128i*)(mask + x));
+#if CV_AVX2
             __m256 mf = _mm256_cmp_ps(_mm256_cvtepi32_ps(_mm256_cvtepi8_epi32(mb)), v_zero, _CMP_NEQ_OQ);
+#else
+            // SSE4.1 expand: AVX-only TUs lack _mm256_cvtepi8_epi32 (AVX2).
+            __m128 mf0 = _mm_cmpneq_ps(_mm_cvtepi32_ps(_mm_cvtepi8_epi32(mb)), _mm_setzero_ps());
+            __m128 mf1 = _mm_cmpneq_ps(_mm_cvtepi32_ps(_mm_cvtepi8_epi32(_mm_srli_si128(mb, 4))), _mm_setzero_ps());
+            __m256 mf = _mm256_insertf128_ps(_mm256_castps128_ps256(mf0), mf1, 1);
+#endif
             __m256 v_src = _mm256_loadu_ps(src + x);
             __m256 v_dst = _mm256_loadu_ps(dst + x);
             __m256 v_new = _mm256_add_ps(_mm256_mul_ps(v_dst, v_beta), _mm256_mul_ps(v_src, v_alpha));
@@ -3383,7 +3392,12 @@ void accW_simd_(const double* src, double* dst, const uchar* mask, int len, int 
         for ( ; x <= len - 4; x += 4)
         {
             __m128i mb = _mm_loadl_epi64((const __m128i*)(mask + x));
+#if CV_AVX2
             __m256d mf = _mm256_castsi256_pd(_mm256_cmpgt_epi32(_mm256_cvtepi8_epi32(mb), _mm256_setzero_si256()));
+#else
+            // SSE4.1 expand: AVX-only TUs lack _mm256_cvtepi8_epi32 / _mm256_cmpgt_epi32 (AVX2).
+            __m256d mf = _mm256_castpd128_pd256(_mm_castsi128_pd(_mm_cmpgt_epi32(_mm_cvtepi8_epi32(mb), _mm_setzero_si128())));
+#endif
             __m256d v_src = _mm256_loadu_pd(src + x);
             __m256d v_dst = _mm256_loadu_pd(dst + x);
             __m256d v_new = _mm256_add_pd(_mm256_mul_pd(v_dst, v_beta), _mm256_mul_pd(v_src, v_alpha));

--- a/modules/imgproc/src/accum.simd.hpp
+++ b/modules/imgproc/src/accum.simd.hpp
@@ -3030,6 +3030,12 @@ void accW_simd_(const ushort* src, float* dst, const uchar* mask, int len, int c
 void accW_simd_(const float* src, float* dst, const uchar* mask, int len, int cn, double alpha)
 {
     int x = 0;
+#if (CV_SIMD || CV_SIMD_SCALABLE) && !defined(CV_ACCUM_FP_USE_YMM)
+    const v_float32 v_alpha = vx_setall_f32((float)alpha);
+    const v_float32 v_beta = vx_setall_f32((float)(1.0f - alpha));
+    const int cVectorWidth = VTraits<v_uint16>::vlanes();
+    const int step = VTraits<v_float32>::vlanes();
+#endif
 #if defined(CV_ACCUM_FP_USE_YMM)
     {
     const __m256 v_alpha = _mm256_set1_ps((float)alpha);
@@ -3069,11 +3075,6 @@ void accW_simd_(const float* src, float* dst, const uchar* mask, int len, int cn
     }
     }
 #elif (CV_SIMD || CV_SIMD_SCALABLE)
-    const v_float32 v_alpha = vx_setall_f32((float)alpha);
-    const v_float32 v_beta = vx_setall_f32((float)(1.0f - alpha));
-    const int cVectorWidth = VTraits<v_uint16>::vlanes();
-    const int step = VTraits<v_float32>::vlanes();
-
     if (!mask)
     {
         int size = len * cn;
@@ -3110,14 +3111,16 @@ void accW_simd_(const float* src, float* dst, const uchar* mask, int len, int cn
             v_store(dst + x + step, v_dst1);
         }
     }
-#endif // CV_SIMD
+#endif // CV_SIMD || CV_SIMD_SCALABLE
 #if (CV_SIMD || CV_SIMD_SCALABLE)
     if (mask && cn == 3)
     {
+#if defined(CV_ACCUM_FP_USE_YMM)
         const v_float32 v_alpha = vx_setall_f32((float)alpha);
         const v_float32 v_beta = vx_setall_f32((float)(1.0f - alpha));
         const int cVectorWidth = VTraits<v_uint16>::vlanes();
         const int step = VTraits<v_float32>::vlanes();
+#endif
         v_uint32 v_0 = vx_setzero_u32();
 
         for ( ; x <= len - cVectorWidth; x += cVectorWidth)

